storage: TestCreateCheckpoint_SpanConstrained can be slow

storage.TestRandomizedIntentInterleavingIter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13090583?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13090583?buildTab=artifacts#/) on master @ [83bcafa004f79bdb156f6921fb8e898df9925c08](https://github.com/cockroachdb/cockroach/commits/83bcafa004f79bdb156f6921fb8e898df9925c08):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestRandomizedIntentInterleavingIter (2s)

```
Stack: 

```
goroutine 5385 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x320
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestRandomizedIntentInterleavingIter
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/df49cd4529be694977d51f20f1abedea/logTestRandomizedIntentInterleavingIter3919216849
    test_log_scope.go:81: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/storage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomizedIntentInterleavingIter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34445

Stressed locally:
```
  Stats over 1000 runs: max = 2.2s, min = 0.8s, avg = 1.5s, dev = 0.2s
  ```

What does the `2s` next to the test name mean here?:
```
panic: test timed out after 4m57s
running tests:
	TestRandomizedIntentInterleavingIter (2s)
```

Is there a timeout above the individual test that's triggering?

--- a/pkg/storage/engine_test.go
+++ b/pkg/storage/engine_test.go
@@ -1082,29 +1082,29 @@ func TestCreateCheckpoint_SpanConstrained(t *testing.T) {
 	ctx := context.Background()
 
 	rng, _ := randutil.NewTestRand()
-	dir, cleanup := testutils.TempDir(t)
-	defer cleanup()
-
 	key := func(i int) roachpb.Key {
 		return keys.SystemSQLCodec.TablePrefix(uint32(i))
 	}
 
+	mem := vfs.NewMem()
+	dir := "foo"
 	db, err := Open(
 		ctx,
-		Filesystem(dir),
+		MakeLocation(dir, mem),
 		cluster.MakeTestingClusterSettings(),
-		TargetFileSize(10<<10 /* 10 KB */),
+		TargetFileSize(2<<10 /* 2 KB */),
 	)
 	assert.NoError(t, err)
 	defer db.Close()
 
 	// Write keys /Table/1/../Table/10000.
+	// 10,000 * 100 byte KVs = ~1MB.
 	b := db.NewWriteBatch()
 	const maxTableID = 10000
 	for i := 1; i <= maxTableID; i++ {
 		require.NoError(t, b.PutMVCC(
 			MVCCKey{Key: key(i), Timestamp: hlc.Timestamp{WallTime: int64(i)}},
-			MVCCValue{Value: roachpb.Value{RawBytes: randutil.RandBytes(rng, 500)}},
+			MVCCValue{Value: roachpb.Value{RawBytes: randutil.RandBytes(rng, 100)}},
 		))
 	}
 	require.NoError(t, b.Commit(true /* sync */))
@@ -1118,17 +1118,19 @@ func TestCreateCheckpoint_SpanConstrained(t *testing.T) {
 		}
 	}
 
-	checkpointRootDir := filepath.Join(dir, "checkpoint")
+	checkpointRootDir := mem.PathJoin(dir, "checkpoint")
 	require.NoError(t, db.FS.MkdirAll(checkpointRootDir, os.ModePerm))
 
 	var checkpointNum int
 	checkpointSpan := func(s roachpb.Span) string {
 		checkpointNum++
-		dir := filepath.Join(checkpointRootDir, fmt.Sprintf("%06d", checkpointNum))
+		dir := mem.PathJoin(checkpointRootDir, fmt.Sprintf("%06d", checkpointNum))
 		t.Logf("Writing checkpoint for span %s to %q", s, dir)
 		assert.NoError(t, db.CreateCheckpoint(dir, []roachpb.Span{s}))
-		assert.DirExists(t, dir)
-		m, err := filepath.Glob(dir + "/*")
+		// Check that the dir exists.
+		_, err := mem.Stat(dir)
+		assert.NoError(t, err)
+		m, err := mem.List(dir)
 		assert.NoError(t, err)
 		assert.True(t, len(m) > 0)
 		t.Logf("Checkpoint wrote files: %s", strings.Join(m, ", "))
@@ -1139,7 +1141,7 @@ func TestCreateCheckpoint_SpanConstrained(t *testing.T) {
 		// Verify that we can open the checkpoint.
 		cDB, err := Open(
 			ctx,
-			Filesystem(dir),
+			MakeLocation(dir, mem),
 			cluster.MakeTestingClusterSettings(),
 			MustExist)
 		require.NoError(t, err)

